net-im/ejabberd-16.08 version bump

candrews opened the pull request:
https://bugs.gentoo.org/show_bug.cgi?id=590502

candrews commented:
@aidecoe how's this look?

aidecoe commented:
Well, it's hard to tell what's the difference between 16.04 and 16.08 from this diff. ;-) I'd expect change in dependencies. It may be needed to bump some of them or add new. Have you compared rebar.config between versions?

I am following ejabberd news and I am aware there are new versions, but I purposefully postpone any bump until I get 16.04 stabilized. See [bug 576398](https://bugs.gentoo.org/show_bug.cgi?id=576398).

candrews commented:
I'm unfortunately not sure how I can make the diff more readable... any suggestions? Would you prefer if I renamed, instead of copied, to create new ebuilds (which of course means the old version is gone, which may not be good)?

I have updated the ejabberd-16.08 ebuild dependencies to match rebar.config. This resulted in discovering that many dependencies didn't have a new enough version in Portage, so I went through and bumped them, too (and for each one, I checked rebar.config and updated those dependencies). I then confirmed that ejabberd emerged and ran.

The only notable thing was that none of the patches are necessary for dev-erlang/lager-3.2.1 as they've all been applied upstream.

I don't think it's normal to bump so many packages in one commit, so I can certainly split each package version bump into a separate commit if you'd like.

SoapGentoo commented on dev-erlang/esip/esip-1.0.7.ebuild:
```diff
@@ -0,0 +1,32 @@
+	rebar_fix_include_path stun
+
+	# ebin contains lonely .gitignore file asking for removal.
+	rm -r "${S}/ebin"
```
`|| die` missing

candrews replied:
Fixed!

Do you have a tool that you run that finds this stuff, or are you doing it by manually reading each line?

SoapGentoo commented on dev-erlang/fast_tls/fast_tls-1.0.6.ebuild:
```diff
@@ -0,0 +1,23 @@
+
+DEPEND=">=dev-erlang/p1_utils-1.0.4
+	>=dev-lang/erlang-17.1
+	dev-libs/openssl:0"
```
missing subslot operator. Would it be possible for you to check libressl support?

candrews replied:
I changed the dependency from `dev-libs/openssl:0` to `dev-libs/openssl:0=`

I also checked libressl support, and that seems to work well, so I've updated the dependency to reflect that too.

SoapGentoo commented on dev-erlang/lager/lager-3.2.1.ebuild:
```diff
@@ -0,0 +1,28 @@
+src_prepare() {
+	rebar_src_prepare
+	# 'priv' directory contains only edoc.css, but doc isn't going to be built.
+	rm -r "${S}/priv"
```
`|| die` missing

SoapGentoo commented:
@candrews While I'm sure parts of it could be automated (like checking whether `|| die` is missing), no, I just look at it.

aidecoe commented:
> I'm unfortunately not sure how I can make the diff more readable... any suggestions? Would you prefer if I renamed, instead of copied, to create new ebuilds (which of course means the old version is gone, which may not be good)?

Bumping is a bit problematic in that matter. File needs to be copied, but on the other hand for a reviewer the difference between these 2 files actually matters (e.g. by providing diff in comment here). It would be also helpful for maintainer to describe how you've made sure that ebuild for a new version is correct. Just bumping version by copying is easier for maintainer than pulling changes from someone else. The actual job is checking whether all dependencies are correct, whether there has been some changes in build scripts that may need to be reflected in ebuild, maybe some extra things have been added (like documentation, some README files, etc). So the job here is proving that everything has been checked - this can be describe in commit message. Also most of dev-erlang packages have got unit tests. Have you run them?

candrews commented:
I have run the test by using this command (which ran successfully):
`FEATURES="test" emerge -1 dev-erlang/cache_tab dev-erlang/esip dev-erlang/fast_tls dev-erlang/fast_xml dev-erlang/fast_yaml dev-erlang/iconv dev-erlang/lager dev-erlang/p1_utils dev-erlang/stringprep dev-erlang/stun net-im/ejabberd`

I have gone through each ebuild and checked the ebuild dependencies against rebar.config and made appropriate changes.

It seems weird to attach a diff as a comment... but, I can do that :)
```
$ diff -u dev-erlang/p1_utils/p1_utils-1.0.*
```

```
$ diff -u dev-erlang/iconv/iconv-1.0.*
```

```
$ diff -u dev-erlang/fast_tls/fast_tls-1.0.*
--- dev-erlang/fast_tls/fast_tls-1.0.3.ebuild   2016-08-04 14:38:11.176125479 -0400
+++ dev-erlang/fast_tls/fast_tls-1.0.6.ebuild   2016-08-05 10:01:43.732952082 -0400
@@ -14,10 +14,12 @@
 LICENSE="Apache-2.0"
 SLOT="0"
 KEYWORDS="~amd64 ~arm ~ia64 ~ppc ~sparc ~x86"
+IUSE="libressl"
 
-DEPEND=">=dev-erlang/p1_utils-1.0.3
+DEPEND=">=dev-erlang/p1_utils-1.0.4
        >=dev-lang/erlang-17.1
-       dev-libs/openssl:0"
+       !libressl? ( dev-libs/openssl:0= )
+       libressl? ( dev-libs/libressl:0= )"
 RDEPEND="${DEPEND}"
 
 DOCS=( CHANGELOG.md  README.md )
```

```
$ diff -u dev-erlang/fast_yaml/fast_yaml-1.0.*
--- dev-erlang/fast_yaml/fast_yaml-1.0.3.ebuild 2016-08-04 14:38:11.176125479 -0400
+++ dev-erlang/fast_yaml/fast_yaml-1.0.5.ebuild 2016-08-04 17:58:41.456366216 -0400
@@ -15,7 +15,7 @@
 SLOT="0"
 KEYWORDS="~amd64 ~arm ~ia64 ~ppc ~sparc ~x86"
 
-DEPEND=">=dev-erlang/p1_utils-1.0.2
+DEPEND=">=dev-erlang/p1_utils-1.0.4
        >=dev-lang/erlang-17.1
        dev-libs/libyaml"
 RDEPEND="${DEPEND}"
```

```
$ diff -u dev-erlang/cache_tab/cache_tab-1.0.*
--- dev-erlang/cache_tab/cache_tab-1.0.2.ebuild 2016-08-04 14:38:11.172792143 -0400
+++ dev-erlang/cache_tab/cache_tab-1.0.3.ebuild 2016-08-04 17:56:16.519762503 -0400
@@ -15,7 +15,7 @@
 SLOT="0"
 KEYWORDS="~amd64 ~arm ~ia64 ~ppc ~sparc ~x86"
 
-DEPEND=">=dev-erlang/p1_utils-1.0.1
+DEPEND=">=dev-erlang/p1_utils-1.0.4
        >=dev-lang/erlang-17.1"
 RDEPEND="${DEPEND}"
 ```

```
$ diff -u dev-erlang/stringprep/stringprep-1.0.*
--- dev-erlang/stringprep/stringprep-1.0.3.ebuild       2016-08-04 14:38:11.176125479 -0400
+++ dev-erlang/stringprep/stringprep-1.0.5.ebuild       2016-08-04 18:00:10.886327984 -0400
@@ -15,7 +15,7 @@
 SLOT="0"
 KEYWORDS="~amd64 ~arm ~ia64 ~ppc ~sparc ~x86"
 
-DEPEND=">=dev-erlang/p1_utils-1.0.3
+DEPEND=">=dev-erlang/p1_utils-1.0.4
        >=dev-lang/erlang-17.1"
 RDEPEND="${DEPEND}"
```

```
$ diff -u dev-erlang/fast_xml/fast_xml-1.1.*
--- dev-erlang/fast_xml/fast_xml-1.1.14.ebuild  2016-08-04 18:05:56.749517644 -0400
+++ dev-erlang/fast_xml/fast_xml-1.1.3.ebuild   2016-08-04 14:38:11.176125479 -0400
@@ -16,7 +16,7 @@
 KEYWORDS="~amd64 ~arm ~ia64 ~ppc ~sparc ~x86"
 IUSE="test"
 
-RDEPEND=">=dev-erlang/p1_utils-1.0.4
+RDEPEND=">=dev-erlang/p1_utils-1.0.0
        >=dev-lang/erlang-17.1
        <dev-lang/erlang-19
        dev-libs/expat"
```

```
$ diff -u dev-erlang/stun/stun-1.0.*
--- dev-erlang/stun/stun-1.0.3.ebuild   2016-08-04 14:38:11.176125479 -0400
+++ dev-erlang/stun/stun-1.0.6.ebuild   2016-08-04 18:00:39.436315874 -0400
@@ -15,8 +15,8 @@
 SLOT="0"
 KEYWORDS="~amd64 ~arm ~ia64 ~ppc ~sparc ~x86"
 
-DEPEND=">=dev-erlang/fast_tls-1.0.0
-       >=dev-erlang/p1_utils-1.0.2
+DEPEND=">=dev-erlang/fast_tls-1.0.6
+       >=dev-erlang/p1_utils-1.0.4
        >=dev-lang/erlang-17.1"
 RDEPEND="${DEPEND}"
```

```
$ diff -u dev-erlang/esip/esip-1.0.*
--- dev-erlang/esip/esip-1.0.4.ebuild   2016-08-04 14:38:11.176125479 -0400
+++ dev-erlang/esip/esip-1.0.7.ebuild   2016-08-05 09:46:28.930010388 -0400
@@ -15,9 +15,9 @@
 SLOT="0"
 KEYWORDS="~amd64 ~arm ~ia64 ~ppc ~sparc ~x86"
 
-DEPEND=">=dev-erlang/fast_tls-1.0.0
-       >=dev-erlang/stun-1.0.0
-       >=dev-erlang/p1_utils-1.0.2
+DEPEND=">=dev-erlang/fast_tls-1.0.6
+       >=dev-erlang/stun-1.0.6
+       >=dev-erlang/p1_utils-1.0.4
        >=dev-lang/erlang-17.1"
 RDEPEND="${DEPEND}"
 
@@ -28,5 +28,5 @@
        rebar_fix_include_path stun
 
        # ebin contains lonely .gitignore file asking for removal.
-       rm -r "${S}/ebin"
+       rm -r "${S}/ebin" || die
 }
```

```
$ diff -u dev-erlang/lager/lager-3.2.*
--- dev-erlang/lager/lager-3.2.0.ebuild 2016-08-04 14:38:11.176125479 -0400
+++ dev-erlang/lager/lager-3.2.1.ebuild 2016-08-05 09:47:37.236647077 -0400
@@ -15,21 +15,14 @@
 SLOT="0"
 KEYWORDS="~amd64 ~arm ~ia64 ~ppc ~sparc ~x86"
 
-DEPEND=">=dev-erlang/goldrush-0.1.7
+DEPEND=">=dev-erlang/goldrush-0.1.8
        >=dev-lang/erlang-17.1"
 RDEPEND="${DEPEND}"
 
 DOCS=( README.md TODO )
-PATCHES=(
-       "${FILESDIR}/0001-Support-typed-records-newly-exposed-in-OTP-19.patch"
-       "${FILESDIR}/0002-Fix-get_env-bug.patch"
-       "${FILESDIR}/0003-Add-get_env-unit-test.patch"
-       "${FILESDIR}/0004-Undo-unnecessary-whitespace.patch"
-       "${FILESDIR}/0005-R15-does-not-appear-to-properly-handle-the-export.patch"
-)
 
 src_prepare() {
        rebar_src_prepare
        # 'priv' directory contains only edoc.css, but doc isn't going to be built.
-       rm -r "${S}/priv"
+       rm -r "${S}/priv" || die
 }
```

```
$ diff -u net-im/ejabberd/ejabberd-16.04.ebuild net-im/ejabberd/ejabberd-16.08.ebuild
--- net-im/ejabberd/ejabberd-16.04.ebuild       2016-08-04 14:38:11.452792248 -0400
+++ net-im/ejabberd/ejabberd-16.08.ebuild       2016-08-04 17:45:36.023385572 -0400
@@ -29,25 +29,24 @@
 # TODO:        >=dev-erlang/moka-1.0.5b
 # TODO: )
 CDEPEND="
-       >=dev-erlang/cache_tab-1.0.2
-       >=dev-erlang/esip-1.0.4
-       >=dev-erlang/fast_tls-1.0.3
-       >=dev-erlang/fast_xml-1.1.3
-       >=dev-erlang/fast_yaml-1.0.3
+       >=dev-erlang/cache_tab-1.0.3
+       >=dev-erlang/esip-1.0.7
+       >=dev-erlang/fast_tls-1.0.6
+       >=dev-erlang/fast_xml-1.1.4
+       >=dev-erlang/fast_yaml-1.0.5
        >=dev-erlang/jiffy-0.14.7
-       >=dev-erlang/lager-3.0.2
+       >=dev-erlang/lager-3.2.1
        >=dev-erlang/luerl-0.2
        >=dev-erlang/p1_oauth2-0.6.1
-       >=dev-erlang/p1_utils-1.0.4
+       >=dev-erlang/p1_utils-1.0.5
        >=dev-erlang/p1_xmlrpc-1.15.1
-       >=dev-erlang/stringprep-1.0.3
-       >=dev-erlang/stun-1.0.3
+       >=dev-erlang/stringprep-1.0.5
+       >=dev-erlang/stun-1.0.6
        >=dev-lang/erlang-17.1[hipe?,odbc?,ssl]
-       <dev-lang/erlang-19
        >=net-im/jabber-base-0.01
        ldap? ( =net-nds/openldap-2* )
        mysql? ( >=dev-erlang/p1_mysql-1.0.1 )
-       nls? ( >=dev-erlang/iconv-1.0.0 )
+       nls? ( >=dev-erlang/iconv-1.0.1 )
        odbc? ( dev-db/unixODBC )
        pam? ( >=dev-erlang/p1_pam-1.0.0 )
        postgres? ( >=dev-erlang/p1_pgsql-1.1.0 )
@@ -63,7 +62,7 @@
 RDEPEND="${CDEPEND}
        captcha? ( media-gfx/imagemagick[truetype,png] )"
 
-PATCHES=( "${FILESDIR}/${P}-ejabberdctl.patch" )
+PATCHES=( "${FILESDIR}/ejabberd-16.04-ejabberdctl.patch" )
 
 EJABBERD_CERT="${EPREFIX}/etc/ssl/ejabberd/server.pem"
 # Paths in net-im/jabber-base
```

aidecoe commented:
````
-PATCHES=( "${FILESDIR}/${P}-ejabberdctl.patch" )
+PATCHES=( "${FILESDIR}/ejabberd-16.04-ejabberdctl.patch" )
```

Why this change?

You have removed `<erlang-19` restriction. Have you tested it builds with newer erlang?

aidecoe commented:
You haven't removed erlang version restriction in fast_xml.

aidecoe commented:
hamcrest, proper, riakc have also version restriction on erlang. These should be checked whether there exist newer versions. ejabberd-16.08 is meant to work with erlang-19, so I suppose all deps are fixed as well in newer versions.

Apart from that it looks fine. Thanks! Diff is really helpful. Otherwise I would just queue this in my mailbox for later to check in some spare time. (-: However... please have in mind I will postpone the pull request until ejabberd-16.04 is stable. Afterwards I will clean up tree from old ejabberd and its old deps.

candrews commented:
```
-PATCHES=( "${FILESDIR}/${P}-ejabberdctl.patch" )
+PATCHES=( "${FILESDIR}/ejabberd-16.04-ejabberdctl.patch" )
```
Because ${P} is now "ejabberd-16.08" and ${FILESDIR}/ejabberd-16.08-ejabberdctl.patch doesn't exist.

dev-erlang/hamcrest looks like a problem as they don't seem to do releases... ever.

dev-erlang/proper released [version 1.2](https://github.com/manopapad/proper/releases/tag/v1.2) which includes R19 support, so I included that bump and dropped the `<dev-lang/erlang-19` dependency.

I ended up bumping dev-erlang/proper-1.2, dev-erlang/protobuffs-0.9.0, dev-erlang/riak_pb-2.1.4.2, and riakc-2.4.1

So if you could take care dev-erlang/hamcrest that should cover all of the <dev-erlang/erlang-19 dependencies.

monsieurp commented:
Please make sure there's no issue generated by the PR.

candrews commented:
@aidecoe dev-erlang/proper-1.2 from this PR has a failing unit test. I'm not enough of a erlang afficionado to know how to fix it or really help upstream figure it out. Could you please take a look? https://github.com/manopapad/proper/issues/129

aidecoe commented:
I have based changes on yours, but I had to correct some things. Please see differences, so you can see what you have missed.

Thank you.

PS. Next time I will bump version sooner.
